application-diagram-api/src/main/java/com/xwiki/diagram/internal/handlers/DiagramContentHandler.java

```diff
@@ -0,0 +1,209 @@
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.diagram.internal.handlers;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParserFactory;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+import org.xwiki.model.reference.DocumentReference;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiAttachment;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * Handler for diagram content operations.
+ * 
+ * @version $Id$
+ * @since 1.13
+ */
+public class DiagramContentHandler
+{
+    /**
+     * View action.
+     */
+    public static final String VIEW_ACTION = "view";
+
+    /**
+     * Pages referenced inside diagram content.
+     */
+    private List<String> includedPages = new ArrayList<String>();
+
+    /**
+     * Helper methods for custom links contained by a diagram.
+     */
+    private DiagramLinkHandler linkHandler;
+
+    /**
+     * Constructor.
+     */
+    public DiagramContentHandler()
+    {
+        this.linkHandler = new DiagramLinkHandler();
+    }
+
+    /**
+     * Update the attachments content and save the new version.
+     * 
+     * @param backlinkDoc document that has a backlink to diagram
+     * @param context the execution context
+     * @param oldDocRef reference of the page before rename
+     * @param newDocRef reference of the page after rename
+     * @throws XWikiException when an error occurs while accessing documents.
+     * @throws IOException if something goes wrong while setting attachment's content.
+     */
+    public void updateAttachment(XWikiDocument backlinkDoc, XWikiContext context, DocumentReference oldDocRef,
```

Comment:
If you make this a component then you don't need to pass the ``XWikiContext`` parameter anymore.